packages/components/src/dashicon/index.js

```diff
@@ -11,6 +16,11 @@
  * @return {JSX.Element} Element
  */
 function Dashicon( { icon, className, ...extraProps } ) {
+	deprecated( '`Dashicon component`', {
+		alternative:
+			'`Icon component` from `@wordpress/icon` package along with `@wordpress/icons` SVG icons or custom SVG icons',
+	} );
```

Comment:
I don't believe this is something can do yet  because of the fact that we still support "string" icons in several places including custom block types. WDYT @gziolo

Reply:
This way we will lose the ability to define icons in `block.json` files. It's rather a bigger change that needs to be reflected in the existing documentation, tutorials, and tools like Create Block. 

I personally don't mind deprecating `Dashicon` component, but we will still need to offer a backward compatible solution for defining icons as strings to ensure all existing 3rd party blocks continue to work.

There is one more thing:

https://github.com/WordPress/gutenberg/blob/master/webpack.config.js#L33

`@wordpress/icons` is still an internal package for WordPress, so people would bundle the same icons in their plugins. Maybe, it isn't a big issue on its own, but it's important to take it into consideration. It also means that folks can't use `wp.icons` global in their code which is different from other functionality WordPress exposes.

Concluding, the code changes proposed are only the very first step in the process that would have to be followed with a good migration recommendation for block authors.